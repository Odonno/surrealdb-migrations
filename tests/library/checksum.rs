use assert_fs::TempDir;
use color_eyre::eyre::{Error, Result};
use insta::{assert_ron_snapshot, Settings};
use itertools::Itertools;
use surrealdb_migrations::MigrationRunner;

use crate::helpers::*;

#[tokio::test]
async fn set_checksum_on_each_migration_file() -> Result<()> {
    let temp_dir = TempDir::new()?;
    let db_name = generate_random_db_name()?;

    add_migration_config_file_with_db_name_in_dir(&temp_dir, DbInstance::Root, &db_name)?;
    scaffold_blog_template(&temp_dir, false)?;

    let config_file_path = temp_dir.join(".surrealdb");

    let configuration = SurrealdbConfiguration {
        db: Some(db_name),
        ..Default::default()
    };

    let db = create_surrealdb_client(&configuration).await?;

    let runner = MigrationRunner::new(&db).use_config_file(&config_file_path);

    runner.up().await?;

    let script_migrations = runner.list().await?;

    let mut insta_settings = Settings::new();
    insta_settings.add_script_timestamp_filter();
    insta_settings.add_datetime_filter();
    insta_settings.bind(|| {
        assert_ron_snapshot!(script_migrations
            .iter()
            .sorted_by(|a, b| Ord::cmp(&b.script_name, &a.script_name))
            .collect_vec());
        Ok::<(), Error>(())
    })?;

    temp_dir.close()?;

    Ok(())
}
