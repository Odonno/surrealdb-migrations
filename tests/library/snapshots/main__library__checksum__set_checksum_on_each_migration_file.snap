---
source: tests/library/checksum.rs
expression: "script_migrations.iter().sorted_by(|a, b|\nOrd::cmp(&b.script_name, &a.script_name)).collect_vec()"
---
[
  ScriptMigration(
    script_name: "[timestamp]_CommentPost",
    executed_at: "[datetime]",
    checksum: Some("9c47faadcf14ea05b44c176a2079ea778eab6db18e4219010f85230508b9ee89"),
  ),
  ScriptMigration(
    script_name: "[timestamp]_AddPost",
    executed_at: "[datetime]",
    checksum: Some("742add6313b4e055d79a9df3651b00b6593ecea8f0c1f7637b3d1f4516e4593f"),
  ),
  ScriptMigration(
    script_name: "[timestamp]_AddAdminUser",
    executed_at: "[datetime]",
    checksum: Some("2a9f8e44975a32f4d1c6e167136c7d9061487c3d86e691f747eb5b8bfd5003a6"),
  ),
]
